sentry-actix/src/lib.rs

```diff
@@ -336,14 +334,13 @@ fn map_status(status: StatusCode) -> protocol::SpanStatus {
     }
 }
 
-/// Build a Sentry request struct from the HTTP request
-fn sentry_request_from_http(request: &ServiceRequest, with_pii: bool) -> (Option<String>, Request) {
-    let transaction = if let Some(name) = request.match_name() {
-        Some(String::from(name))
-    } else {
-        request.match_pattern()
-    };
+/// Extract a transaction name from the HTTP request
+fn transaction_name_from_http(req: &ServiceRequest) -> String {
+    format!("{} {}", req.method(), req.uri())
```

Comment:
I believe `req.uri()` might have too high of a cardinality and might even contain PII.

I believe `match_pattern` is the thing we want, is it documented to give you `/user/{id}/profile` for example: https://docs.rs/actix-web/latest/actix_web/struct.HttpRequest.html#method.match_pattern

`match_name` seems to refer to the function used for logging? Not sure, there is no example.

Adding a `req.method()` prefix does seem like a very good thing indeed.

Reply:
Agree 100%.

I've changed the code to use `request.match_pattern()`, fixed the broken tests and added an extra one that does a POST with a pattern.